src/engine/enginebuffer.cpp

```diff
@@ -1310,18 +1310,15 @@ void EngineBuffer::postProcess(const int iBufferSize) {
     }
     const mixxx::Bpm localBpm = m_pBpmControl->updateLocalBpm();
     double beatDistance = m_pBpmControl->updateBeatDistance();
-    // FIXME: Double check if calling setLocalBpm with an invalid value is correct and intended.
-    mixxx::Bpm newLocalBpm;
     if (localBpm.isValid()) {
-        newLocalBpm = localBpm;
-    }
-    m_pSyncControl->setLocalBpm(newLocalBpm);
-    SyncMode mode = m_pSyncControl->getSyncMode();
-    m_pSyncControl->reportPlayerSpeed(m_speed_old, m_scratching_old);
-    if (isLeader(mode)) {
-        m_pEngineSync->notifyBeatDistanceChanged(m_pSyncControl, beatDistance);
-    } else if (isFollower(mode)) {
-        m_pSyncControl->updateTargetBeatDistance();
+        m_pSyncControl->setLocalBpm(localBpm);
+        SyncMode mode = m_pSyncControl->getSyncMode();
+        m_pSyncControl->reportPlayerSpeed(m_speed_old, m_scratching_old);
+        if (isLeader(mode)) {
```

Comment:
if localbpm is not valid and isLeader is true, we should change sync mode to Follower.

Reply:
done & force-pushed.